projects/Mallard/src/services/iap.ts

```diff
@@ -0,0 +1,60 @@
+import RNIAP, { Purchase } from 'react-native-iap'
+import { Platform } from 'react-native'
+import { ITUNES_CONNECT_SHARED_SECRET } from 'src/constants'
+
+export interface ReceiptIOS {
+    expires_date: string
+    expires_date_ms: string
+    expires_date_pst: string
+    is_in_intro_offer_period: string
+    is_trial_period: string
+    original_purchase_date: string
+    original_purchase_date_ms: string
+    original_purchase_date_pst: string
+    original_transaction_id: string
+    product_id: string
+    purchase_date: string
+    purchase_date_ms: string
+    purchase_date_pst: string
+    quantity: string
+    transaction_id: string
+    web_order_line_item_id: string
+}
+
+const fetchDecodeReceipt = (receipt: string) =>
+    RNIAP.validateReceiptIos(
+        {
+            'receipt-data': receipt,
+            password: ITUNES_CONNECT_SHARED_SECRET,
+        },
+        __DEV__,
+    )
+
+const getMostRecentTransactionReceipt = (purchases: Purchase[]) => {
+    if (!purchases.length) return false
+    return purchases.sort((a, b) => b.transactionDate - a.transactionDate)[0]
+        .transactionReceipt
+}
+
+const isReceiptActive = (receipt: ReceiptIOS) => {
+    const expirationInMilliseconds = Number(receipt.expires_date_ms)
+    const nowInMilliseconds = Date.now()
+    return expirationInMilliseconds > nowInMilliseconds
+}
+
+// The essence of this came from here: https://github.com/dooboolab/react-native-iap/issues/275#issuecomment-433582389
+const fetchActiveIOSSubscriptionReceipt = async (): Promise<ReceiptIOS | null> => {
```

Comment:
would it be 'cleaner' to reject the promise here rather than return null?

Reply:
The problem with that is that for genuine exceptions we're currently showing an error but not finding a receipt is a "happy path" and we handle things differently: based on whether they return null or whether they throw. Potentially we could change this though as actually they may not need to take a separate path, but I'll have a look at it and not change it here as it would balloon this PR. Good point though ...